testsuite/module-publish/src/packages/complex/sources/vector_picture.move

```diff
@@ -86,13 +96,30 @@ module 0xABCD::vector_picture {
         let all_palettes = borrow_global<AllPalettes>(palette_addr);
         let palette_addr = vector::borrow(&all_palettes.all, palette_index);
 
-        let palette_ = borrow_global_mut<Palette>(*palette_addr);
+        let palette = borrow_global_mut<Palette>(*palette_addr);
 
         // Confirm the index is not out of bounds.
-        assert!(index < vector::length(&palette_.vec), error::invalid_argument(E_INDEX_OUT_OF_BOUNDS));
+        assert!(index < vector::length(&palette.vec), error::invalid_argument(E_INDEX_OUT_OF_BOUNDS));
 
         // Write the pixel.
         let color = Color { r, g, b };
-        *vector::borrow_mut(&mut palette_.vec, index) = color;
+        *vector::borrow_mut(&mut palette.vec, index) = color;
+    }
+
+    public entry fun check(
+        palette_addr: address,
+        palette_index: u64,
+        index: u64,
+    ) acquires Palette, AllPalettes {
+        let all_palettes = borrow_global<AllPalettes>(palette_addr);
+        let palette_addr = vector::borrow(&all_palettes.all, palette_index);
+
+        let palette = borrow_global_mut<Palette>(*palette_addr);
+
+        // Confirm the index is not out of bounds.
+        assert!(index < vector::length(&palette.vec), error::invalid_argument(E_INDEX_OUT_OF_BOUNDS));
+
+        let color = vector::borrow(&mut palette.vec, index);
+        assert!(color.r != 255 || color.g != 255 || color.b != 255, error::invalid_argument(E_MAX_COLOR));
```

Comment:
What is the purpose of this?

Reply:
so that we actually use the value, and move compiler cannot optimize it out (at some point in the future, I don't think it does that now :) )